ros_ws/src/teleoperation/include/keyboard_controller.h

```diff
@@ -1,37 +1,87 @@
 #pragma once
 
-#include <ros/ros.h>
-
+#include <SDL2/SDL.h>
+#include <algorithm>
+#include <array>
 #include <drive_msgs/drive_param.h>
-#include <std_msgs/Int64.h>
-
+#include <ros/package.h>
+#include <ros/ros.h>
 #include <signal.h>
-#include <termios.h>
-#include <time.h>
+#include <stdexcept>
 
-#define TOPIC_DRIVE_PARAMETERS "/set/drive_param"
+constexpr const char* TOPIC_DRIVE_PARAMETERS = "/set/drive_param";
 
 enum class Keycode : int
 {
-    W     = 119,
-    A     = 97,
-    S     = 115,
-    D     = 100,
+    W = 119,
+    A = 97,
+    S = 115,
+    D = 100,
     SPACE = 32
 };
 
+enum class KeyIndex : int
+{
+    ACCELERATE = 0,
+    DECELERATE = 2,
+    STEER_LEFT = 1,
+    STEER_RIGHT = 3
+};
+
+constexpr std::array<Keycode, 4> KEY_CODES = { Keycode::W, Keycode::A, Keycode::S, Keycode::D };
+
+constexpr int KEY_COUNT = 4;
```

Comment:
You could move the `contexpr int KEY_COUNT` above the  `constexpr std::array<Keycode, 4>` and use the `constexpr` instead of the number. `KEY_COUNT` could also be used in ` std::array<bool, 4> m_key_pressed_state` instead of the `4`.